src/store/calendarObjectInstance.js

```diff
@@ -1351,6 +1351,61 @@ const mutations = {
 			}
 		}
 	},
+
+	addAttachmentBySharedData(state, { calendarObjectInstance, sharedData }) {
+		const attachment = AttachmentProperty.fromLink(sharedData.url)
+		const fileName = sharedData.path
+
+		// hot-fix needed temporary, becase calendar-js has no fileName get-setter
+		const parameterFileName = new Parameter('FILENAME', fileName)
+		const managedId = new Parameter('MANAGED-ID', `${sharedData.file_source}||${sharedData.has_preview}`)
```

Comment:
Custom 
```suggestion
		const managedId = new Parameter('X-MANAGED-ID', `${sharedData.file_source}||${sharedData.has_preview}`)
```

Non-standard parameters should be prefixed with `X-`.

Reply:
It's https://datatracker.ietf.org/doc/html/rfc8607#section-4.3

Reply:
Oh, right. I'm sorry for my stupidity.

But still, I don't like the custom serialization format. Ideally we should use base64 encoded JSON data and handle the encoding/serialization transparently through calendar-js.

Reply:
@st3iny we can use parameters (X-NC-PREVIEW X-NC-FILE-ID), then this will simplify event processing and their correct selection in the future. now I have found a bug related to the slpit method. I'll redo it.

Reply:
https://github.com/nextcloud/calendar/pull/4251#issuecomment-1207938079